stdlib/runtime/numbers.gr

```diff
@@ -2508,6 +2508,88 @@ provide let (>>) = (x: Number, y: Number) => {
 // [NOTE]: Coercion is a *conservative* process! For example, even if a float is 1.0,
 //         we will fail if attempting to coerce to an int!
 
+/**
+ * Converts a Number to an Int32.
+ *
+ * @param number: The value to convert
+ * @returns The Number represented as an Int32
+ *
+ * @since v0.2.0
+ */
+@unsafe
+provide let coerceNumberToShortUint = (x: Number, max32, max64, is8bit) => {
+  let x = WasmI32.fromGrain(x)
+  let int32 = if (isSimpleNumber(x)) {
+    untagSimple(x)
+  } else {
+    let xtag = boxedNumberTag(x)
+    match (xtag) {
+      t when WasmI32.eq(t, Tags._GRAIN_INT32_BOXED_NUM_TAG) => {
+        boxedInt32Number(x)
+      },
+      t when WasmI32.eq(t, Tags._GRAIN_INT64_BOXED_NUM_TAG) => {
+        let int64 = boxedInt64Number(x)
+        if (WasmI64.gtS(int64, max64) || WasmI64.ltS(int64, 0N)) {
+          throw Exception.Overflow
+        }
+        WasmI32.wrapI64(int64)
+      },
+      t when WasmI32.eq(t, Tags._GRAIN_BIGINT_BOXED_NUM_TAG) => {
+        BI.toInt32(x)
+      },
+      _ => {
+        // rationals are never integral, and we refuse to coerce floats to ints
+        throw Exception.NumberNotIntlike
+      },
+    }
+  }
+  if (WasmI32.gtS(int32, max32) || WasmI32.ltS(int32, 0n)) {
+    throw Exception.Overflow
+  }
+  if (is8bit) WasmI32.and(int32, 0xffffn) else WasmI32.and(int32, 0xffffn)
+}
+
+@unsafe
+provide let coerceNumberToShortInt =
```

Comment:
No docblock?

Reply:
I actually don't think one is required here since it never gets propagated up to the stdlib; same goes for the one above now that I think of it